### PR TITLE
feat: add monitoring guide

### PR DESCRIPTION
**Pull request by ggera**
## fixes https://github.com/KILTprotocol/ticket/issues/1606
docker-compose script to setup monitoring for a collator
## How to test:
Follow the steps and be able to setup monitoring with little help

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: docs/chain/collator.md [hunk 1]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
 
 ## Monitoring
 
-TODO
+It would be ideal if monitoring is done from a different host however in cases of resource constraints it would be better to run these monitoring tools on the same machine where your collator is running than not having.The following Monitoring stack is recommended.
```
Comment (weichweich) [inline docs/chain/collator.md]: For better git diffs we try to always have a single sentence per line.

```suggestion
It would be ideal if monitoring is done from a different host however in cases of resource constraints it would be better to run these monitoring tools on the same machine where your collator is running than not having monitoring at all.
The following Monitoring stack is recommended.
```

File: docs/chain/collator.md [hunk 2]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+You can either run grafana by your own or subscribe free cloud option here https://grafana.com/products/cloud/.
```
Comment (weichweich) [inline docs/chain/collator.md]: I always like to have a link that describes itself. (i.e. I prefer: "[a nice blockchain project](https://kilt.io)" over "a nice blockchain project ([here](https://kilt.io))")

```suggestion
You can either run grafana by yourself or use the [hosted service of grafana](https://grafana.com/products/cloud/). 
```

File: docs/chain/collator.md [hunk 3]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
 
 ## Monitoring
 
-TODO
+It would be ideal if monitoring is done from a different host however in cases of resource constraints it would be better to run these monitoring tools on the same machine where your collator is running than not having.The following Monitoring stack is recommended.
+You can either run grafana by your own or subscribe free cloud option here https://grafana.com/products/cloud/. 
+NB: if you want to use cloud option you need to expose prometheus to outside access by chaning the ports section to ```  - "9090:9090"``` and reverse proxying with webserver is recommended.
```
Comment (weichweich) [inline docs/chain/collator.md]: * What does `NB` stand for?

```suggestion
NB: if you want to use cloud option you need to expose prometheus to outside access by changing the ports section to `- "9090:9090"` and reverse proxying with web server is recommended.
```

File: docs/chain/collator.md [hunk 4]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+- Slack
+
+There are two types of metrics we are going to collect. **Node Metrics** and **Blockchain Metrics.**
+
+### Prerequisite
+```
+sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+sudo chmod +x /usr/local/bin/docker-compose
+sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+```
```
Comment (weichweich) [inline docs/chain/collator.md]: Might be best to link to the official docker compose guide? We would need to update if there is a new version or anything.

File: docs/chain/collator.md [hunk 5]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+Steps:
+
+1. Clone or download our monitoring template from this link  https://github.com/KILTprotocol/docs
```
Comment (weichweich) [inline docs/chain/collator.md]: you can add links with `[text...](https://link-to-website)`

File: docs/chain/collator.md [hunk 6]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+2. Change directory to the above cloned project's collator directory: ```cd docs/collator```
+3. Edit .env file to have your change grafana admin password
+4. Run the following command if you want to install only prometheus and node exporter
+   ```./start-collator.sh```
```
Comment (weichweich) [inline docs/chain/collator.md]: No need to put that in a separate line. It will be formatted as one line anyways

```suggestion
4. Run the following command if you want to install only prometheus and node exporter `./start-collator.sh`
```

File: docs/chain/collator.md [hunk 7]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+5. If you want to install prometheus with grafana comment line 6 of ``` start-collator.sh``` and uncomment line 7 then run step 4
+6. if you want to install prometheus, grafana and your collator with docker-compose comment line 6  of ``` start-collator.sh``` and uncomment line 8 then run  step 4
```
Comment (weichweich) [inline docs/chain/collator.md]: i think that all the references to lines etc are quite hard to follow. Also if we change the files this will all change too. We will most likely change the files without updating the guide and than it's even harder to follow.

File: docs/chain/collator.md [hunk 8]
```diff
@@ -405,7 +405,40 @@ The corresponding methods can be found as an extrinisc under `parachainStaking >
...
+
+### Testing
+You could open and check HOST_IP:3000  then authenticate with admin:ADMIN_PASSWORD you set in .env at step 3.
+
+### Configuring Notification channel
+Choose anyone of the notification channels and follow this documentation so that you get alerts and notifications when there are critical alerts
+https://grafana.com/docs/grafana/latest/alerting/old-alerting/notifications/
```
Comment (weichweich) [inline docs/chain/collator.md]: please use `[text...](https://link-to-website)` an inline link.

File: collator/.env [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+ADMIN_USER=admin
+ADMIN_PASSWORD=@dm1nPa55wd
+
+
```
Comment (weichweich) [inline collator/.env]: We only need an empty line at the end. Also don't provide a default password. IMHO The user must set the password.

```suggestion
ADMIN_USER=admin
ADMIN_PASSWORD=
```

File: collator/grafana/provisioning/dashboards/host_machine.json [hunk 10]
```diff
@@ -0,0 +1,1441 @@
+{
```
Comment (weichweich) [inline collator/grafana/provisioning/dashboards/host_machine.json]: maybe we could just point them to the node-exporter-full dashboard? That would make the guide easier.

File: collator/grafana/provisioning/dashboards/spiritnet.json [hunk 11]
```diff
@@ -0,0 +1,1548 @@
...
+          "mode": "single"
+        }
+      },
+      "pluginVersion": "8.0.2",
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "rate(substrate_rpc_calls_total[5m])",
+          "interval": "",
+          "legendFormat": "{{statefulset_kubernetes_io_pod_name}}",
```
Comment (weichweich) [inline collator/grafana/provisioning/dashboards/spiritnet.json]: this will only work in kubernetes? Please check that the dashboard works with the docker compose setup.

File: collator/grafana/provisioning/dashboards/spiritnet.json [hunk 12]
```diff
@@ -0,0 +1,1548 @@
...
+  },
+  "time": {
+    "from": "now-30m",
+    "to": "now"
+  },
+  "timepicker": {},
+  "timezone": "",
+  "title": "spiritnet",
+  "uid": "JGBmHZI7k",
+  "version": 1
```
Comment (weichweich) [inline collator/grafana/provisioning/dashboards/spiritnet.json]: These should not be there.

```suggestion
  "title": "spiritnet"
```

File: collator/start-collator.sh [hunk 13]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+
+docker-compose up -d  #Prometheus and node exporter are
```
Comment (weichweich) [inline collator/start-collator.sh]: I think we don't need a script for a single command?

File: collator/stop-collator.sh [hunk 14]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -e
+set +x
+
+docker-compose down
```
Comment (weichweich) [inline collator/stop-collator.sh]: I think we don't need a script for a single command?

**Review by weichweich — CHANGES_REQUESTED** [review]
I have few requests.

File: collator/grafana/provisioning/dashboards/host_machine.json (hunk 10 above)
Comment (ggera) [reply collator/grafana/provisioning/dashboards/host_machine.json]: Merged the dashboards for simiplicity

File: collator/start-collator.sh (hunk 13 above)
Comment (ggera) [reply collator/start-collator.sh]: Removed

File: collator/stop-collator.sh (hunk 14 above)
Comment (ggera) [reply collator/stop-collator.sh]: Removed

File: collator/grafana/provisioning/dashboards/spiritnet.json (hunk 12 above)
Comment (ggera) [reply collator/grafana/provisioning/dashboards/spiritnet.json]: I keep then so that I can point them to the intial dashboard by going http://localhost:3000/d/JGBmHZI7k/kilt-spiritnet?orgId=1&refresh=10s

File: collator/docker-compose.yml [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    restart: unless-stopped
+    user: "1000"
+    ports:
+      # - "9944:9944"
+      - "30336:30336"
+      - "30333:30333"
+    volumes:
+      - ~/data:/data
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (weichweich) [inline collator/docker-compose.yml]: this should not be required anymore?

File: collator/docker-compose.yml [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+      - /:/rootfs:ro
+    command:
+      - '--path.procfs=/host/proc'
+      - '--path.rootfs=/rootfs'
+      - '--path.sysfs=/host/sys'
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    ports:
+      - "9100"
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (weichweich) [inline collator/docker-compose.yml]: this should not be required anymore?

File: collator/docker-compose.yml [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+    ports:
+      - "9090:9090"
+    user: "root"
+    volumes:
+      - ./prometheus.yml:/etc/prometheus/prometheus.yml
+      - ./auth_config.yml:/etc/config/auth_config.yml
+      - ~/data/prometheus:/prometheus
+    command: --config.file=/etc/prometheus/prometheus.yml --web.config.file=/etc/config/auth_config.yml
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (weichweich) [inline collator/docker-compose.yml]: this should not be required anymore?

**Review by weichweich — APPROVED** [review]
we don't need the host internal extra_host thing anymore right?

File: collator/docker-compose.yml (hunk 15 above)
Comment (ggera) [reply collator/docker-compose.yml]: Removed
